pkg/gardenlet/controller/shoot/shoot/reconciler_migrate.go

```diff
@@ -158,13 +162,23 @@ func (r *Reconciler) runMigrateShootFlow(ctx context.Context, o *operation.Opera
 		// in case it has changed. Also, it's needed for other control plane components like the kube-apiserver or kube-
```

Comment:
do we need workerless in  `cleanupShootResources` here also?

Reply:
> Will there be shoot state created for a workerless resource?

Yes, shoot state will be there for workerless shoot also, shoot state holds some secrets also like etcd-ca, observability secrets and etc.

Reply:
>  If you look at the shoot state resource that gets created, it has worker, infrastructure details among others. In addition you have steps like RestoreControlPlane, migrateInfrastructure, deleteInfrastructure which one should relook at in context to workerless shoots

Can you please explain the concern here?